pkg/apis/autoscaling/v1alpha1/hvpa_types.go

```diff
@@ -172,21 +237,84 @@ const (
 	In Scaling = "in"
 )
 
+// LastError has detailed information of the error
+type LastError struct {
+	// Description of the error
+	Description string `json:"description,omitempty"`
+
+	// Time at which the error occurred
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty"`
+
+	// LastOperation is the type of operation for which error occurred
+	LastOperation string `json:"lastOperation,omitempty"`
+}
+
 // HvpaStatus defines the observed state of Hvpa
 type HvpaStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-	// current information about the HPA.
-	// +optional
-	HpaStatus HpaStatus `json:"hpaStatus,omitempty" protobuf:"bytes,1,opt,name=hpaStatus"`
+	// Replicas is the number of replicas of the target resource.
+	Replicas *int32 `json:"replicas,omitempty"`
+	// HpaSelector is the string form of the label selector of HPA. This is required for HPA to work with scale subresource.
+	HpaSelector *string `json:"hpaSelector,omitempty"`
+	// Current HPA UpdatePolicy set in the spec
+	HpaUpdatePolicy *UpdatePolicy `json:"hpaUpdatePolicy,omitempty"`
+	// Current VPA UpdatePolicy set in the spec
+	VpaUpdatePolicy *UpdatePolicy `json:"vpaUpdatePolicy,omitempty"`
 
-	// Current information about the VPA.
-	// +optional
-	VpaStatus vpa_api.VerticalPodAutoscalerStatus `json:"vpaStatus,omitempty" protobuf:"bytes,2,opt,name=vpaStatus"`
+	HpaWeight VpaWeight `json:"hpaWeight,omitempty"`
+	VpaWeight VpaWeight `json:"vpaWeight,omitempty"`
 
-	// Current information about HVPA.
-	// +optional
-	HvpaStatus HvpaCurrentStatus `json:"hvpaCurrentStatus,omitempty" protobuf:"bytes,3,opt,name=hvpaCurrentStatus"`
+	// Override scale up stabilization window
+	OverrideScaleUpStabilization bool `json:"overrideScaleUpStabilization,omitempty"`
+
+	LastBlockedScaling *[]*BlockedScaling
```

Comment:
```suggestion
	LastBlockedScaling []*BlockedScaling `json:"lastBlockedScaling,omitempty"`
```

No need for pointers to slices. Slices are references already.
Beware of the whitespace if you are going to apply this change automatically :-)